src/build/GhosttyResources.zig

```diff
@@ -0,0 +1,257 @@
+        const install_step = b.addInstallDirectory(.{
+            .source_dir = upstream.path("ghostty"),
+            .install_dir = .{ .custom = "share" },
+            .install_subdir = b.pathJoin(&.{ "ghostty", "themes" }),
+            .exclude_extensions = &.{".md"},
+        });
+        try steps.append(&install_step.step);
+    }
+
+    // Fish shell completions
+    {
+        const wf = b.addWriteFiles();
+        _ = wf.add("ghostty.fish", buildpkg.fish_completions);
+
+        const install_step = b.addInstallDirectory(.{
+            .source_dir = wf.getDirectory(),
+            .install_dir = .prefix,
+            .install_subdir = "share/fish/vendor_completions.d",
+        });
+        try steps.append(&install_step.step);
+    }
+
+    // zsh shell completions
+    {
+        const wf = b.addWriteFiles();
+        _ = wf.add("_ghostty", buildpkg.zsh_completions);
+
+        const install_step = b.addInstallDirectory(.{
+            .source_dir = wf.getDirectory(),
+            .install_dir = .prefix,
+            .install_subdir = "share/zsh/site-functions",
+        });
+        try steps.append(&install_step.step);
+    }
+
+    // bash shell completions
+    {
+        const wf = b.addWriteFiles();
+        _ = wf.add("ghostty.bash", buildpkg.bash_completions);
+
+        const install_step = b.addInstallDirectory(.{
+            .source_dir = wf.getDirectory(),
+            .install_dir = .prefix,
+            .install_subdir = "share/bash-completion/completions",
+        });
+        try steps.append(&install_step.step);
+    }
+
+    // Vim plugin
+    {
+        const wf = b.addWriteFiles();
+        _ = wf.add("syntax/ghostty.vim", config_vim.syntax);
+        _ = wf.add("ftdetect/ghostty.vim", config_vim.ftdetect);
+        _ = wf.add("ftplugin/ghostty.vim", config_vim.ftplugin);
+        _ = wf.add("compiler/ghostty.vim", config_vim.compiler);
+
+        const install_step = b.addInstallDirectory(.{
+            .source_dir = wf.getDirectory(),
+            .install_dir = .prefix,
+            .install_subdir = "share/vim/vimfiles",
+        });
+        try steps.append(&install_step.step);
+    }
+
+    // Neovim plugin
+    // This is just a copy-paste of the Vim plugin, but using a Neovim subdir.
+    // By default, Neovim doesn't look inside share/vim/vimfiles. Some distros
+    // configure it to do that however. Fedora, does not as a counterexample.
+    {
+        const wf = b.addWriteFiles();
+        _ = wf.add("syntax/ghostty.vim", config_vim.syntax);
+        _ = wf.add("ftdetect/ghostty.vim", config_vim.ftdetect);
+        _ = wf.add("ftplugin/ghostty.vim", config_vim.ftplugin);
+        _ = wf.add("compiler/ghostty.vim", config_vim.compiler);
+
+        const install_step = b.addInstallDirectory(.{
+            .source_dir = wf.getDirectory(),
+            .install_dir = .prefix,
+            .install_subdir = "share/nvim/site",
+        });
+        try steps.append(&install_step.step);
+    }
```

Comment:
Seems like we code do some code-reuse here if you wanted. Might also be useful in their own files. Would it be useful to add a build option for vim/neovim support files (probably also bat), which defaulted to true?

Reply:
We could but I think now that its in a separate file it's pretty clean to have them split out like this (wet, not dry) because it makes it very clear what each thing is doing. It's all simple code. 

We could now very easily add build flags to control what is included but unless someone actually has a use case for that I don't think we have to yet.